src/main/java/ro/unibuc/triplea/infrastructure/game/repository/GameRepositoryImpl.java

```diff
@@ -0,0 +1,132 @@
+package ro.unibuc.triplea.infrastructure.game.repository;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Primary;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import ro.unibuc.triplea.application.auth.dto.response.GameResponse;
+import ro.unibuc.triplea.domain.auth.model.entity.Game;
+import ro.unibuc.triplea.domain.auth.repository.GameRepository;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+@Primary
+public class GameRepositoryImpl implements GameRepository {
+
+    @Value("${STEAM_API_KEY}")
+    private String steamApiKey;
+    
+    private final RestTemplate restTemplate;
+    private final SpringDataGameRepository springDataGameRepository;
+
+    public GameRepositoryImpl(RestTemplate restTemplate, SpringDataGameRepository springDataGameRepository) {
+        this.restTemplate = restTemplate;
+        this.springDataGameRepository = springDataGameRepository;
+    }
+
+    @Override
+    public List<GameResponse> findAll(Optional<Integer> count) {
```

Comment:
findAll is an improper name for this function because you provide a count as argument. Maybe another name should be fine. findAll is a proper name for a function that finds all names without needing a counting argument

Reply:
renamed with findGames